guppylang/prelude/angles.py

```diff
@@ -1,96 +1,68 @@
 """Guppy standard module for dyadic rational angles."""
 
-# mypy: disable-error-code="empty-body, misc, override"
+# mypy: disable-error-code="empty-body, misc, override, operator"
 
+import math
 from typing import no_type_check
 
-from hugr import tys as ht
 from hugr import val as hv
+from hugr.std.float import FloatVal
 
 from guppylang.decorator import guppy
-from guppylang.prelude._internal.checker import CoercingChecker
-from guppylang.prelude._internal.compiler.angle import AngleOpCompiler
-from guppylang.prelude.builtins import nat
+from guppylang.module import GuppyModule
+from guppylang.prelude.builtins import py
 
-_hugr_angle_type = ht.Opaque("angle", ht.TypeBound.Copyable, [], "tket2.quantum")
+angles = GuppyModule("angles")
 
 
-def _hugr_angle_value(numerator: int, log_denominator: int) -> hv.Value:
-    custom_const = {
-        "log_denominator": log_denominator,
-        "value": numerator,
-    }
-    return hv.Extension(
-        name="ConstAngle",
-        typ=_hugr_angle_type,
-        val=custom_const,
-        extensions=["quantum.tket2"],
-    )
-
-
-pi = guppy.constant("pi", ty="angle", value=_hugr_angle_value(1, 1))
-
-
-@guppy.type(_hugr_angle_type)
+@guppy.struct(angles)
 class angle:
-    """The type of angles represented as dyadic rational multiples of 2π."""
-
-    @guppy.custom(AngleOpCompiler("afromrad"), CoercingChecker())
-    def __new__(radians: float) -> "angle": ...
-
-    @guppy.custom(AngleOpCompiler("aadd"))
-    def __add__(self: "angle", other: "angle") -> "angle": ...
-
-    @guppy.custom(AngleOpCompiler("asub"))
-    def __sub__(self: "angle", other: "angle") -> "angle": ...
-
-    @guppy.custom(AngleOpCompiler("aneg"))
-    def __neg__(self: "angle") -> "angle": ...
+    halfturns: float
```

Comment:
Doc-comment that this (is misnamed) is not an angle in the truest sense but a rotation, i.e. does not identify with itself modulo 2pi (/2halfturns)

Reply:
Done in #506